packages/experimental/Badge/src/useBadge.ts

```diff
@@ -10,3 +12,30 @@ export const useBadge = (props: BadgeProps): BadgeProps => {
     ...rest,
   };
 };
+
+export type BadgeColors = {
+  backgroundColor?: ColorValue;
+  color?: ColorValue;
+  iconColor?: ColorValue;
+  borderColor?: ColorValue;
+  backgroundColorDark?: ColorValue;
+  colorDark?: ColorValue;
+  borderColorDark?: ColorValue;
+  hcBackground?: ColorValue;
+  hcColor?: ColorValue;
+  hcBorderColor?: ColorValue;
+};
+
+export function getHCProps(theme: Theme, colors?: BadgeColors) {
+  const hcColors = colors || {};
+  const hcBackground = hcColors.hcBackground || theme.colors.transparentBackground;
+  const hcColor = hcColors.hcColor || theme.colors.neutralForeground3;
+  const hcBorderColor = hcColors.hcBorderColor || theme.colors.neutralForeground3;
+
+  return {
+    backgroundColor: hcBackground,
+    color: hcColor,
+    iconColor: hcColor,
+    borderColor: hcBorderColor,
+  };
+}
```

Comment:
Not sure if useBadge is a good place for these methods. But I had to move it because of this warning:
![image](https://user-images.githubusercontent.com/11574680/184182772-6bf5c5e6-e8d3-439c-a41d-fb14b956448c.png)

Reply:
You can put the function in a new file if there isn't a good place for it.